File: my_py_pkg/my_py_pkg/number_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

# Import data type
from example_interfaces.msg import Int64

class NumberPublisherNode(Node): 
    def __init__(self):
        super().__init__("number_publisher")

        self.declare_parameter("number_to_publish", 9)
        self.declare_parameter("publish_frequency", 1.0)
        
        self.number_ = self.get_parameter("number_to_publish").value
        self.publish_frequency_ = self.get_parameter("publish_frequency").value

        # Create publisher obj
        self.publisher_ = self.create_publisher(Int64, "number", 10) # type, topic name, queue size

        # add timer
        self.timer_ = self.create_timer(1.0/self.publish_frequency_, self.publish_number)

        self.get_logger().info("Number Publisher has been started")

    def publish_number(self):
        # Create msg obj
        msg = Int64()
        # Add custom data
        msg.data = self.number_
        # Output the data
        self.publisher_.publish(msg)
        self.get_logger().info("Message published!")


def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisherNode() 
    rclpy.spin(node)
    rclpy.shutdown()


if __name__=="__main__":
    main()

File: my_cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

class NumberPublisherNode : public rclcpp::Node 
{
public:
    NumberPublisherNode() : Node("number_publisher") 
    {
        this->declare_parameter("number_to_publish", 2);
        this->declare_parameter("publish_frequency", 1.0);
        this->number_ = this->get_parameter("number_to_publish").as_int();
        this->publish_frequency_ = this->get_parameter("publish_frequency").as_double();
        publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number",10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds((int)(1000.0/this->publish_frequency_)), std::bind(&NumberPublisherNode::publishNumber, this));
        RCLCPP_INFO(this->get_logger(), "Number Publisher has been started.");
    }

    void publishNumber()
    {
        auto msg = example_interfaces::msg::Int64();
        msg.data = this->number_;
        publisher_->publish(msg);
    }

private:
    int number_;
    double publish_frequency_;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberPublisherNode>();   
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}